### ISSUE TITLE
ValueError: Using a target size (torch.Size([32])) that is different to the input size (torch.Size([8320])) is deprecated. Please ensure they have the same size.

### ISSUE THREAD

**Issue by lcslima45** [V-Sher/GANs_Pytorch#2]
Hi I'm testing your tutorial in my own custom data and I don't know if the version of Pytorch I'm using is different from yours but when I go to the step of TRAINING GAN step of the tutorial I get an error inside the line of the loss_function

'''
loss_real = loss(yhat, target)
''' 

I got the error in the title of issue

More details below 

'''
ValueError                                Traceback (most recent call last)
<ipython-input-41-d93c23fecf02> in <module>()
     21         target = torch.ones(len(b), dtype=torch.float, device=device)
     22         # calculate loss
---> 23         loss_real = loss(yhat, target)
     24         # calculate gradients -  or rather accumulation of gradients on loss tensor
     25         loss_real.backward()

2 frames
/usr/local/lib/python3.7/dist-packages/torch/nn/functional.py in binary_cross_entropy(input, target, weight, size_average, reduce, reduction)
   2884         raise ValueError(
   2885             "Using a target size ({}) that is different to the input size ({}) is deprecated. "
-> 2886             "Please ensure they have the same size.".format(target.size(), input.size())
   2887         )
   2888 

ValueError: Using a target size (torch.Size([32])) that is different to the input size (torch.Size([8320])) is deprecated. Please ensure they have the same size.
'''

**Comment by V-Sher** [comment]
I doubt it has anything to do with the Pytorch version but in case it helps, I just tested it with the latest one and it works fine. 

Coming back to the error, it looks like the shape of your `yhat` and `target` tensors are not aligned. Ideally, they should be the same (try `assert yhat.shape == target.shape`). Guessing by the error message, the target size i.e. `(torch.Size([32]))` seems correct since we set the batch size to 32 so that's how many elements should be present in the 1-D target tensor. The input tensor i.e. `yhat` doesn't seem right and looks like you didn't set the dataloader properly, because of which batch `b` is not the right shape. Right now it looks like you are sending all 8320 images at once. Without looking at the code, that's my best guess. If it isn't resolved, I suggest uploading the code so we can work through it together.

Good luck!
